src/localplayer.cpp

```diff
@@ -639,86 +639,86 @@ void LocalPlayer::applyControl(float dtime)
 	if (continuous_forward)
 		speedH += move_direction;
 
-	if (control.up) {
-		if (continuous_forward) {
-			if (fast_move)
-				superspeed = true;
-		} else {
-			speedH += move_direction;
+	if (m_knockback_timer > 0.0f) {
+		m_knockback_timer -= dtime;
+		if (m_knockback_timer <= 0.0f) {
+			m_speed = v3f(0, -9.81 * BS, 0);
 		}
-	}
-	if (control.down) {
-		speedH -= move_direction;
-	}
-	if (!control.up && !control.down) {
-		speedH -= move_direction *
-			(control.forw_move_joystick_axis / 32767.f);
-	}
-	if (control.left) {
-		speedH += move_direction.crossProduct(v3f(0,1,0));
-	}
-	if (control.right) {
-		speedH += move_direction.crossProduct(v3f(0,-1,0));
-	}
-	if (!control.left && !control.right) {
-		speedH -= move_direction.crossProduct(v3f(0,1,0)) *
-			(control.sidew_move_joystick_axis / 32767.f);
-	}
-	if(control.jump)
-	{
-		if (free_move) {
-			if (g_settings->getBool("aux1_descends") || always_fly_fast) {
+	} else {
```

Comment:
No need to block other inputs, when a player is in mid-jump or is falling, they can still move, this should happen during knockback too.